### PR TITLE
Avoid E2E teardown retries after browser closes

### PR DESCRIPTION
**Pull request by alfonso-noriega**
## What

Stops E2E app teardown from retrying after Playwright has already closed the browser page, and caps best-effort dashboard app deletion to a fixed teardown time budget.

## Why

Recent failed runs show repeated teardown noise with the same browser-closed pattern:

- 36/80 recent failed runs matched E2E teardown/browser-close cleanup errors
- 27/80 included `Target page, context or browser has been closed`
- 27/80 included `app deletion failed after 3 attempts`

A related failure mode is the app deploy E2E completing its assertions but then consuming the remaining test timeout while trying to delete a secondary app from the Dev Dashboard. If dashboard cleanup cannot complete quickly, the per-test cleanup scripts can reap the orphan later.

## Testing

- `pnpm --filter e2e lint`
- `pnpm --filter e2e type-check` attempted previously, but blocked by existing `cli-kit:build` errors for missing GraphiQL dependencies (`h3`, `react-dom/server`, `@shopify/polaris`, `@shopify/polaris-icons`).

### CODE REVIEW

**Comment by alfonso-noriega** [comment]
* **#7803** <a href="https://app.graphite.com/github/pr/Shopify/cli/7803?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/Shopify/cli/7803?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/e2e/setup/app.ts [hunk 1]
```diff
@@ -337,7 +344,8 @@ export async function findAppOnDevDashboard(page: Page, appName: string, orgId?:
 
     // Check for next page
     const nextLink = page.locator('a[href*="next_cursor"]').first()
-    if (!(await isVisibleWithin(nextLink, BROWSER_TIMEOUT.medium))) break
+    const remainingMs = deadline ? Math.max(deadline - Date.now(), 0) : BROWSER_TIMEOUT.medium
```
Comment (isaacroldan) [inline packages/e2e/setup/app.ts:347]: 🐛 **Bug**: When the dashboard search reaches the deadline after scanning links, `remainingMs` is clamped to `0` and passed to `isVisibleWithin`. Playwright treats `timeout: 0` as disabling the timeout, so a missing pagination link can wait until the enclosing test/browser timeout instead of returning `null` and letting teardown stop. This reintroduces the teardown hang this change is intended to cap.

**Suggestion:** Return immediately when `remainingMs <= 0` before calling `isVisibleWithin`, or ensure the timeout passed to Playwright is never `0` and use an explicit deadline check such as `Date.now() >= deadline`.
